### PR TITLE
Merging the latest Giac, fixing the build system

### PR DESCRIPTION
**Pull request by gn-novak-n**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "files.associations": {
+        "tommath.h": "c"
```
Comment (gn-novak-n) [inline .vscode/settings.json]: remove file

**Comment by gn-novak-n** [comment]
So far we were relying on the builds coming from two repos - a fork of the official Giac repo and a private repo that is containing all of the custom tweaks and API wrapper. The idea was likely to be able to do clean sync with the Giac in the fork and than overwrite what needs to be overwritten form the customization repo. However, since we have been intervening in the .cpp code itself (e.g. by exposing the Timeout functionality), updating the Giac in the forked repo is a non-trivial task requiring careful merging. So, the purpose of having the two repos becomes void. Going forward, we will likely simplify our life, move to a single repo containing all of what we need to build the Giac framework, at the expense of more difficult merging.

This PR is an intermediate step in that direction, we are still not checking in everything from the customization repo, but we do want to check in e.g. *.gradle files.

File: src/giac/cpp/subst.cc [hunk 2]
```diff
@@ -3606,10 +3636,49 @@ namespace giac {
     }
     return simplest;
   }
+#else
+  gen _trigsimplify(const gen & g,GIAC_CONTEXT) {
```
Comment (Hugo-TH-Chu) [inline src/giac/cpp/subst.cc:3640]: Great to see we have trigsimplify in the PR!

**Review by Hugo-TH-Chu — APPROVED** [review]
LGTM. Thanks Novak! Since my C++ knowledge is practically nil, I'll let @trunggnsap be the judge.

**Pull request merged**